common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -643,8 +725,13 @@ public void registerExecutor(String appId, ExecutorShuffleInfo executorInfo) {
             AppShuffleInfo appShuffleInfo = originalAppShuffleInfo.get();
             logger.warn("Cleanup shuffle info and merged shuffle files for {}_{} as new " +
                 "application attempt registered", appId, appShuffleInfo.attemptId);
-            mergedShuffleCleaner.execute(
-              () -> closeAndDeletePartitionFilesIfNeeded(appShuffleInfo, true));
+            // Clean up all the merge shuffle related information in the DB for the former attempt
+            submitCleanupTask(
+              () -> {
+                removeAppAttemptPathInfoFromDB(new AppAttemptId(appId, appShuffleInfo.attemptId));
```

Comment:
Move this `removeAppAttemptPathInfoFromDB` into `appsShuffleInfo.compute` - before `writeAppPathsInfoToDb` (if `appShuffleInfo != null`) - this will keep the DB consistent w.r.t `appsShuffleInfo`.

Reply:
Updated to make strong consistent between DB and in memory hashmap.